### PR TITLE
Feature: Ability for user to provide a custom azure environment 

### LINKED ISSUE #2050: Provide ability to specify custom azure environment
Right now public, china, germany and us cloud are hard coded. We should provide ways for user to specify a custom cloud

### PR DESCRIPTION
**Pull request by timotheeguerin**
fix #2050

image

image

### CODE REVIEW

**Comment by bgklein** [comment]
Nit: maybe if no custom environments specify Add instead of Edit.

Also Im assuming the friendly name part is one you added with mock data and not a default right?

**Comment by timotheeguerin** [comment]
Good idea, yeah when you click edit the fist time in generate the file and that's the default values.

**Comment by timotheeguerin** [comment]
Won't show up if you never edit

File: src/client/startup.ts [hunk 1]
```diff
@@ -56,11 +57,17 @@ async function startApplication(batchExplorerApp: BatchExplorerApplication, menu
     });
 }
 
-export async function startBatchExplorer() {
+export async function startBatchExplorer(args: BatchExplorerArgs) {
     // Those warnings are electron complaining we are loading remote data
     // But this is a false positive when using dev server has it doesn't seem to ignore localhost
     process.env.ELECTRON_DISABLE_SECURITY_WARNINGS = "true";
 
+    log.info("PArams", args);
```
Comment (timotheeguerin) [inline src/client/startup.ts]: remove

**Pull request merged**